### ISSUE TITLE
OnWorkListener missing after SDK update from 3.2.1 to 3.3.0 ?!!

### ISSUE THREAD

**Issue by MehdiBrahmi** [at-internet/piano-analytics-android#5]
Hello here 👋

We use the `OnWorkListener` interface to capture the body of the event sent and display it in a logger screen (in our pre-production application).
This feature allows our QA testers to check the content of each event directly from the application without needing to install Android Studio + App Inspection or Charles Proxy.

```kotlin
private val onWorkListener = OnWorkListener {
    override fun onBeforeSend(built: BuiltModel?, stored: MutableMap<String, BuiltModel>?): Boolean {
        built?.let { logEvent(it.body, it.uri) }
        return super.onBeforeSend(built, stored)
    }
}

/* ... */

fun send(name: String, data: Map<String, Any>) {
    pianoAnalytics.sendEvent(Event(name, data), null, onWorkListener)
}

```

When i update SDK from 3.2.1 to 3.3.0 i was surprise to see that this interface does not exist anymore 🤔 ?!!

Can you provide us the new method for retrieving the request body ?

I'd like to add a small comment about this update. Upgrading from version 3.2.1 to version 3.3.0 should not cause any breaking changes. It might be interesting to follow a standard rules such as [Semantic Versioning](https://semver.org/) like :

> Given a version number MAJOR.MINOR.PATCH, increment the:
> 
> MAJOR version when you make incompatible API changes
> MINOR version when you add functionality in a backward compatible manner
> PATCH version when you make backward compatible bug fixes
> Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.

Thanks in advance.

**Comment by MehdiBrahmi** [comment]
Hello DeKan, the aim is for our testers to validate the accuracy of the events we send. The testers are not developers and don't have a logcat console. So we're looking for a way to allow them to validate the data with a log screen in our application (on an internal flavor).

As I said last week, the `OnWorkListener` listener was really useful for retrieving the event body and url. Can you reinstate something similar?

Thanks :)

**Comment by DeKaN** [comment]
Hi @MehdiBrahmi ,

I mean what do you prefer to see: the first or the second solution?
```
// Callback, which will be called after all events' processors. You can't change events here
PianoAnalytics.getInstance().eventProcessorCallback = EventProcessorCallback  { events -> }

// 1
fun interface EventProcessorCallback {
    fun onProcess(events: List<String>) // list of event JSON data
}
// 2
fun interface EventProcessorCallback {
    fun onProcess(events: List<Event>) // list of `io.piano.android.analytics.model.Event` objects 
}
```
I think the second option is better for your case

**Comment by MehdiBrahmi** [comment]
Hi @DeKaN
The first one is more useful, because I want the data to be as close as possible to what's returned and I've seen, for example, that in your code, when you serialise  an Object to a JSON String, you rename the `properties` attribute of the `Event` object to `data`.

is it possible to have something like before with the json object and the url ? In fact, the url contains a query parameter with the visitor's ID, enabling our testers to find their session in the analytics dashboard and validate our tagging plan.

Thank you for helping ;)

**Comment by DeKaN** [comment]
Hi @MehdiBrahmi 
I'm trying to add a flexible solution, that can be used by any client. And I think the second solution is better, because: anyone can decide how to display event data. 

For example, somebody will print to logcat via:
```
        PianoAnalytics.getInstance().eventProcessorCallback = PianoAnalytics.EventProcessorCallback { events ->
            events.forEach { event ->
                val eventData = event.properties.joinToString(prefix = "[", postfix = "]") { p -> "${p.name.key}=${p.value}" }

                // Visitor ID here is always actual, it will be used at sending event if offlineStorageMode != OfflineStorageMode.ALWAYS
                Timber.d("Visitor ID = '${pa.visitorId}', event name = '${event.name}', event data = '$eventData'")
            }
        }
```
this code prints text like 
```
Visitor ID = '<...>', event name = 'page.display', event data = '[page_full_name=main, app_fs=false, app_fsau=false, app_sc=4, app_dsls=0, app_dsfs=0, app_fsd=20230718, app_sessionid=<...>, device_screen_width=720, device_screen_height=1361, os_group=android, os_version=10, os=android 10, device_manufacturer=HMD Global, device_model=Nokia 5.1 Plus, device_timestamp_utc=1689673569, browser_language=en, browser_language_local=US, connection_type=WIFI, event_collection_platform=android, event_collection_version=3.3.0, app_id=com.example.piano_analytics_android, app_version=1.0, visitor_privacy_mode=optin, visitor_privacy_consent=true]'
```

Or you can write RecyclerView Adapter, which shows event name in "collapsed" state and event name with properties list in  "expanded" state

**Comment by DeKaN** [comment]
Released with version 3.3.1
